File: backend/LeadApi/Models/Lead.cs
namespace LeadApi.Models;

public class Lead
{
    public Guid Id { get; set; }
    public string Nome { get; set; } = string.Empty;
    public string Empresa { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Telefone { get; set; } = string.Empty;
    public string Status { get; set; } = "novo";
    public string Origem { get; set; } = string.Empty;
    public DateTimeOffset CriadoEm { get; set; }
    public string Campanha { get; set; } = string.Empty;
    public string ProdutoInteresse { get; set; } = string.Empty;
    public decimal ValorEstimado { get; set; }
    public string? ObservacoesInternas { get; set; }

    public ICollection<LeadComment> Comentarios { get; set; } = new List<LeadComment>();
    public ICollection<LeadTimelineEntry> Historico { get; set; } = new List<LeadTimelineEntry>();
}

File: src/types/lead.ts
export type LeadStatus = 'novo' | 'em_andamento' | 'convertido' | 'perdido';

export interface LeadSummary {
  id: string;
  nome: string;
  empresa: string;
  email: string;
  telefone: string;
  status: LeadStatus;
  origem: string;
  criadoEm: string;
}

export interface LeadDetail extends LeadSummary {
  campanha: string;
  produtoInteresse: string;
  valorEstimado: number;
  observacoesInternas?: string;
  comentarios: LeadComment[];
  historico: LeadTimelineEntry[];
}

export interface LeadComment {
  id: string;
  autor: string;
  conteudo: string;
  criadoEm: string;
}

export interface LeadTimelineEntry {
  id: string;
  descricao: string;
  criadoEm: string;
  criadoPor: string;
}

export interface LeadFilter {
  busca?: string;
  status?: LeadStatus | 'todos';
  origem?: string;
  dataInicial?: string;
  dataFinal?: string;
}

export interface LeadUpdatePayload {
  status?: LeadStatus;
  observacoesInternas?: string;
}

export interface CommentPayload {
  conteudo: string;
}
